_posts/2023-02-27-generating-your-own-fonts-with-fantasicon.md

```diff
@@ -0,0 +1,148 @@
+---
+layout: post
+title: "Generating your own fonts with Fantasicon"
+date: 2023-02-27 15:51:42
+tags: fonts, icons, design
+categories:
+twitter_text: "Generating your own fonts with Fantasicon"
+authors: Kevin Hougasian
+image: https://images.unsplash.com/photo-1609605348579-3123e3d40eb8?ixlib=rb-4.0.3&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1674&q=80
+image_url: https://unsplash.com/photos/92-mTYj5oGs
+image_credit: Brett Jordan
+---
+
+Anyone who's heard me mention [Platform UI](https://platformui.com/) probably knows we make our own font icons, and bake them right into Platform UI, to keep it a one-stop resource. There are some wonderful font packages out there, but you don't always want to load thousands of icons to get the particular mix you're going for. 
+
+For anyone interested, here what I found along the way...
+
+## The SVGs
+
+I'm not going to go in-depth on generating the SVGs. Here's our basic setup:
+
+| source | default |
+|---|---|
+| artboard | 4" x 4" |
+| main stroke | 24pt |
+| auxiliary stroke | 18pt |
+| corner | .15" |
+| join |	round |
+
+
+There are a few tools out there you can evaluate for your workflow. We currently use SVGO, but will incorporate outlie-stroke in a future build.
+
+ - https://github.com/svg/svgo 
+ - https://github.com/elrumordelaluz/outline-stroke 
+ - https://github.com/svgdotjs/svg.js
+
+
+## Ok, let's make some icons
+
+Our first attempt was ok. It did the job but was pretty basic and it gererated the font everytime.
+
+ ```js
+module.exports = {
+  inputDir: './src/optimized-icons',
+  outputDir: './src/generated',
+  fontTypes: ['ttf', 'woff', 'woff2'],
+  assetTypes: ['scss'],
+  prefix: 'pi',
+  name: "platform-icons"
+};
+``` 
+Fontasticon has a great [Readme](https://github.com/tancredi/fantasticon#readme) and [Discord channel](https://discord.gg/BXAY3Kc3mp). I couldn't have done this without BOTH of them! 
+
+A big issue we faced was adding new icons - we would overwrite previous values making the use of `:before` and `:after` impossible. Enter `codepoints`; But how did they really work? 
+
+Let walk through our current config file `.fantasticonrc.js` to see how it all comes together. We use [Vite](https://vitejs.dev/) to generate Platform UI (with icons) and a stand-alone Platform-icons repo.
```

Comment:
```suggestion
Let's walk through our current config file `.fantasticonrc.js` to see how it all comes together. We use [Vite](https://vitejs.dev/) to generate Platform UI (with icons) and a stand-alone Platform-icons repo.
```

Reply:
yeah, I did not run this through Grammarly :(